value/src/main/java/com/google/auto/value/processor/gwtserializer.vm

```diff
@@ -17,14 +17,13 @@ package $pkg;
 #foreach ($i in $imports)
 import $i;
 #end
+import com.google.gwt.user.client.rpc.CustomFieldSerializer;
 import com.google.gwt.user.client.rpc.SerializationException;
 import com.google.gwt.user.client.rpc.SerializationStreamReader;
 import com.google.gwt.user.client.rpc.SerializationStreamWriter;
 
 @${generated}("com.google.auto.value.processor.AutoValueProcessor")
-public final class $serializerClass {
-  // There are no instances of this class.
-  private ${serializerClass}() {}
+public final class $serializerClass$formalTypes extends CustomFieldSerializer<$subclass$actualTypes> {
```

Comment:
It's an additional (optional) enhancement to reduce the amount of reflection at runtime on server side.

When GWT-RPC finds a custom field deserializer class, if it extends `CustomFieldSerializer`, it'll instantiate it and then call the instance methods; otherwise it'll use reflection to get the static methods and call them. So extending `CustomFieldSerializer` reduce the amount of reflection needed.
There's also a `ServerCustomFieldSerializer` for _type-checked_ deserialization; IIUC this is mostly needed for generic classes, so that could be another enhancement step for auto-value.

See http://www.gwtproject.org/javadoc/latest/com/google/gwt/user/client/rpc/CustomFieldSerializer.html and http://www.gwtproject.org/doc/latest/DevGuideServerCommunication.html#DevGuideCustomSerialization